lock-api-objects/src/main/java/com/palantir/lock/LockDescriptor.java

```diff
@@ -74,10 +75,10 @@ public byte[] getBytes() {
 
     @Override
     public int hashCode() {
-        final int prime = 31;
-        int result = 1;
-        result = prime * result + Arrays.hashCode(bytes);
-        return result;
+        if (hashCode == 0) {
+            hashCode = 31 + Arrays.hashCode(bytes);
+        }
+        return hashCode;
```

Comment:
`LockDescriptor` is used in many hash based `Set`s, notably in `ConjureTimelockResource`

https://github.com/palantir/atlasdb/blob/31f015c747daffc57499dad665ba0c56a0f60596/timelock-impl/src/main/java/com/palantir/atlasdb/timelock/ConjureTimelockResource.java#L116-L158